lib/middleware.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var middleware = require('express').Router();
+
+module.exports = function generateMiddleware(options) {
+
+  options = options || {};
+
+  options['cookie-name'] = options['cookie-name'] || 'hof-cookie-check';
+
+  middleware.get('/cookies-required', function renderCookiesRequired(req, res) {
+    res.render('cookies-required');
+  });
+
+  middleware.get('/check-cookies', function checkCookie(req, res) {
+    if (Object.keys(req.cookies).length && req.cookies[options['cookie-name']]) {
+      res.redirect(req.cookies[options['cookie-name']]);
```

Comment:
I don't know what crazy fool came up with the idea of putting the URL inside the cookie. But can we trust it?
Could a malicious script alter the cookie and mislead the user in some way? Do we care? Should we encrypt the cookie or maybe only store the path rather than the full URL with the domain?